Test to ensure documents are indexed properly

niconosenzo opened the pull request:
https://bugzilla.redhat.com/show_bug.cgi?id=1575662#c5

richm commented:
/retest

richm commented:
@niconosenzo Does the latest failure mean the bug has not been fixed in the actual code?  That is, this isn't a test failure, it is the test correctly reporting a problem in the logging code?

nhosoi commented on test/indexing_after_ns_removal.sh:
```diff
@@ -64,13 +75,38 @@ done
+    [ "$?" != "0" ] && os::log::error ${namespace_uid} does not match any UID collected from the actual namespace list during namespace creation
+
+    pod_uid=$(curl_es $es_pod /"${index}"/_search?q=kubernetes.namespace_id:* | jq . | grep -i pod_id | awk -F'"' '{ print $4 }')
+    grep -m 1 "${pod_uid}" temp_ns_uid &>/dev/null
```
s/temp_ns_uid/temp_pod_uid/ ???

niconosenzo replied:
'Yes' to both of your comments.

richm commented:
/retest

richm commented:
@jcantrill This test failure means the fluent-plugin-kubernetes is not handling the namespace create/delete/create case correctly - i.e. there is still a bug in that plugin.

nhosoi commented on test/indexing_after_ns_removal.sh:
```diff
@@ -64,13 +75,38 @@ done
+
+
+# .orphaned index  is not created by default, so if it exists, then it contains records
+[[ $(curl_es $es_pod /_cat/indices?h=index,docs.count | grep .orphaned) ]] && os::log::error  There are records in the .orphaned index.
```
Found one issue.  This last line returns 1 if .orphaned is not found.  That is, if successful, this test ends with (failure == 1); if .orphaned is found, (success == 0).
```
$ [[ $(cat /etc/hosts | grep .orphaned) ]] && echo "found"
$ echo $?
1
$ [[ $(cat /etc/hosts | grep localhost) ]] && echo "found"
found
$ echo $?
0
```
Can we do "if .orphaned is found, print error; exit 1; else exit 0"?

niconosenzo replied:
@nhosoi Done, did some tests and throwing the expected result now.

niconosenzo commented:
- Removed es_pod references (it was returning  non-zero exit code)
- Orphan indices check fixed.
- Fixed main while loop, now the it waits till pod is "Completed" to remove it.
- Adding "|| true" to uid matching loop to avoid failing early

niconosenzo commented:
/retest

richm commented:
/retest

richm commented:
/retest

richm commented:
/retest

richm commented:
/retest

jcantrill commented:
Closing as stale and most of these tests have since moved into the operator(s) code base
